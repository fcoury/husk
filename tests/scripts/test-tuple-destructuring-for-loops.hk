// Test tuple destructuring in for loops

// Test basic tuple destructuring
let pairs = [(1, "one"), (2, "two"), (3, "three")];

println("=== Testing Basic Tuple Destructuring ===");
for (number, word) in pairs {
    println(format!("{}: {}", number, word));
}

// Test coordinate processing
let coordinates = [(0, 0), (10, 20), (30, 40)];

println("=== Testing Coordinate Processing ===");
for (x, y) in coordinates {
    println(format!("Point at ({}, {})", x, y));
}

// Test mixed data types
let mixed = [(1, "hello"), (2, "world")];

println("=== Testing Mixed Data Types ===");
for (id, text) in mixed {
    println(format!("ID {}: {}", id, text));
}

println("All tuple destructuring tests completed!");