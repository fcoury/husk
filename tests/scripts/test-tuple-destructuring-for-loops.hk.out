=== Testing Basic Tuple Destructuring ===
1: one
2: two
3: three
=== Testing Coordinate Processing ===
Point at (0, 0)
Point at (10, 20)
Point at (30, 40)
=== Testing Mixed Data Types ===
ID 1: hello
ID 2: world
All tuple destructuring tests completed!
